testing/deployer/topology/compile.go

```diff
@@ -238,10 +242,9 @@ func compile(logger hclog.Logger, raw *Config, prev *Topology) (*Topology, error
 			}
 			addTenancy(n.Partition, "default")
 
-			if _, exists := seenNodes[n.ID()]; exists {
-				return nil, fmt.Errorf("cannot have two nodes in the same cluster %q with the same name %q", c.Name, n.ID())
+			if n.IsServer() && n.Partition != "default" {
```

Comment:
Adding this restriction. Practically this would have failed in consul itself, but this makes it fail up higher where it's more actionable.